src/main/kotlin/com/hibob/academy/resource/petsResource.kt

```diff
@@ -0,0 +1,55 @@
+package com.hibob.academy.resource
+
+import com.hibob.academy.types.Pet
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.RequestBody
+import java.util.*
+
+@Controller
+@Path("/api/noy/pets")
+@Produces(MediaType.APPLICATION_JSON)
+class PetsResource {
+
+// create
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun addPet(@RequestBody pet: Pet): Response {
+        Response.status(Response.Status.CREATED).build()
+        return Response.ok(pet).build()
+    }
+
+// update
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/updatePetName/{petId}")
```

Comment:
So just like the other paths you did `/{petId}/doSomething` this would also be the case here.

So the route at the end is read like - from all the pets `/pets/` pick 1 - with id `/{petId}` and update it's name `/updateName` or `/updatePetName`.